src/main/java/teammates/storage/entity/Account.java

```diff
@@ -25,6 +28,9 @@
 
     private String institute;
 
+    @Unindex
+    private Map<String, Long> readNotifications;
```

Comment:
1. Instead of using `Long`, we can use `Instant`, and let `InstantTranslatorFactory` help us translate to `TimestampValue` when writing to the cloud datastore. 
2. Probably can consider using [`@Serialize`](https://www.javadoc.io/doc/com.googlecode.objectify/objectify/latest/com/googlecode/objectify/annotation/Serialize.html) on the `readNotifications` too.

Reply:
From what I understand, the translation of `Instant` to `TimestampValue` can be done when the property itself is of type `Instant`. However, in this case, the property type is `Map`, so I would likely need to write a new `ValueTranslatorFactory` to translate `Map<String, Instant>` to `Map<String, TimestampValue>`. Given that, would it be better to create a new `ValueTranslator`, or store the timestamp as `Long` and convert to `Instant` only when using the timestamp (e.g. to check for notification expiry)?

As for `@Serialize`, seems that it might cause issues where changing the attribute would cause past data to be unreadable, not sure if it should be used here?

Reply:
> From what I understand, the translation of `Instant` to `TimestampValue` can be done when the property itself is of type `Instant`. 
Oh yes you are right, `InstantTranslatorFactory` doesn't help us here, sorry about that.

> However, in this case, the property type is `Map`, so I would likely need to write a new `ValueTranslatorFactory` to translate `Map<String, Instant>` to `Map<String, TimestampValue>`. 
If we use `@Serialize` we probably can avoid this (https://github.com/objectify/objectify/wiki/Entities#serializing). 
 
> As for `@Serialize`, seems that it might cause issues where changing the attribute would cause past data to be unreadable, not sure if it should be used here?
Do you mean when we update the attribute? How so? 🤔

Reply:
My bad! I was looking this part of the documentation where it states that any changes to classes will prevent past objects from being deserialized on fetch:
<img width="919" alt="Screenshot 2022-03-20 at 18 19 20" src="https://user-images.githubusercontent.com/54243224/159157832-179d73a5-60b2-4d49-a3c8-889637dccc94.png">
But since`HashMap` is serializable by default, it won't cause an issue, sorry for the confusion caused!

In that case, I'll change `readNotifications` from `Map<String, Long>` to `Map<String, Instant>` and serialize it 👍